examples/step-22/step-22.cc

```diff
@@ -216,37 +218,40 @@ namespace Step22
 
 
   // We implement similar functions for the right hand side which for the
-  // current example is simply zero:
+  // current example is simply zero. Since we have the incompressibility
+  // constraint, the dim-th component of the right hand side will always be 0
+  // and therefore we return a dim dimensional Tensor which represents
+  // only the external forces $\textbf f$.
   template <int dim>
-  class RightHandSide : public Function<dim>
+  class RightHandSide : public TensorFunction<1, dim>
   {
   public:
     RightHandSide()
-      : Function<dim>(dim + 1)
+      : TensorFunction<1, dim>(dim)
```

Comment:
The meaning of the argument you pass here is that of a time :-) So better go this way:
```suggestion
      : TensorFunction<1, dim>()
```
Or, alternatively, just omit the constructor altogether.